### PR TITLE
Loocv updates

### PR DESCRIPTION
**Pull request by jelmerborst**
Loocv:
- loocv consolidation to handle backward models, single models and condition based models
- new loocv_func that does loocv correctly: it applies a provided func to the data of n-1 subjects, and then calculates the likelihood on the left out subject. This function can be as complex as you want and also test multiple models at once. It's also not much slower in the end.

Cross collleration:
- fixed using numpy crosscorr method. Works now independent of template width and shape

In addition, while doing these a bunch of small fixes which only happen in edge cases, in particular:
- remove nans on pmfs
- min_iteration >= 1 instead of >
- checking for -inf

### CODE REVIEW

File: src/hsmm_mvpy/models.py
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:810]: Why the np.isneginf(lkh)? I didn't have this situation before I think

File: src/hsmm_mvpy/models.py
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:969]: We agreed on this but somehow this is a strange fix, I'll investigate after the merge

File: src/hsmm_mvpy/models.py
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:1103]: I never checked, does this happen regularly?

File: src/hsmm_mvpy/models.py
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:1458]: Awesome :)

File: src/hsmm_mvpy/simulations.py
Comment (GWeindel) [inline src/hsmm_mvpy/simulations.py:63]: On the sfreq it probably needs a default or a catch for is None in line 135

Comment (GWeindel) [reply src/hsmm_mvpy/simulations.py:63]: I would say we can just go for a default at 100Hz

File: src/hsmm_mvpy/simulations.py
Comment (GWeindel) [inline src/hsmm_mvpy/simulations.py:176]: Yes rounding here won't hurt

File: src/hsmm_mvpy/utils.py
Comment (GWeindel) [inline src/hsmm_mvpy/utils.py:668]: we should probably lower-case this function for consistency, I shouldn't have created it in uppercase...

File: src/hsmm_mvpy/utils.py
Comment (GWeindel) [inline src/hsmm_mvpy/utils.py:1]: Everything looks absolutely great! We might have to keep an eye as the change in the code base are quite numerous but we'll see this on the fly after the merge

**Review by GWeindel — CHANGES_REQUESTED** [review]
That is a great PR! I attached some reviews but nothing major, I think we can merge to main very soon

File: src/hsmm_mvpy/models.py
Comment (jelmerborst) [reply src/hsmm_mvpy/models.py:810]: This only happened in a very specific situation (feeding in already optimal mags/params, still going through EM), but I don't think it hurts. It might not be necessary with the i >= min_iteration, but I wasn't entirely sure.

File: src/hsmm_mvpy/models.py
Comment (jelmerborst) [reply src/hsmm_mvpy/models.py:1103]: No, again only in an edge case where it tries to define super narrow pdfs. This fixes it, but then makes the likelihood go to -inf (otherwise you get errors)

File: src/hsmm_mvpy/simulations.py
Comment (jelmerborst) [reply src/hsmm_mvpy/simulations.py:63]: good point, yes 100hz would be good here!

File: src/hsmm_mvpy/simulations.py
Comment (jelmerborst) [reply src/hsmm_mvpy/simulations.py:176]: not necessary after all.

File: src/hsmm_mvpy/utils.py
Comment (jelmerborst) [reply src/hsmm_mvpy/utils.py:668]: then it needs a different name, I will fix it.

File: src/hsmm_mvpy/utils.py
Comment (jelmerborst) [reply src/hsmm_mvpy/utils.py:1]: sounds like a plan. Let's merge after my next commit, then run the tutorials and prep the next workshop to see where things break.

File: src/hsmm_mvpy/models.py
Comment (GWeindel) [reply src/hsmm_mvpy/models.py:810]: Yes it can stay, I was just curious

File: src/hsmm_mvpy/models.py
Comment (GWeindel) [reply src/hsmm_mvpy/models.py:1103]: We can leave it to avoid the errors but then make a more clear warning, but not necessary right now

**Review by GWeindel — APPROVED** [review]
All green!

**Pull request merged**
